src/extension.ts

```diff
@@ -47,6 +49,69 @@ export function activate(context: ExtensionContext) {
     })
   );
 
+  // When extension is started, loop for each text documents which are opened to update their language ID.
+  workspace.textDocuments.forEach(document => {
+    updateLanguageId(document);
+  });
+  // When a text document is opened,  update their language ID.
+  context.subscriptions.push(
+    workspace.onDidOpenTextDocument((document) => {
+      updateLanguageId(document);
+    })
+  );
+
+  /**
+   * Update if required the language ID to 'quarkus-properties' if needed.
+   *
+   * @param document the text document.
+   */
+  async function updateLanguageId(document: TextDocument) {
+    const fileName: string = path.basename(document.fileName);
+    if (fileName === 'application.properties') {
+      if (document.languageId === 'quarkus-properties') {
+        // the language ID is already quarkus-properties do nothing.
+        return;
+      }
+      tryToForceLanguageId(document, fileName, 'quarkus-properties');
+    } else if (fileName === 'application.yaml' || fileName === 'application.yml') {
+      if (document.languageId === 'yaml') {
+        // the language ID is already yaml do nothing.
+        return;
+      }
+      tryToForceLanguageId(document, fileName, 'yaml');
+    }
+  }
+
+  /**
+   * Try to force the document language ID to the given language ID.
+   *
+   * @param document the text document.
+   * @param languageId the language ID.
+   */
+  async function tryToForceLanguageId(document: TextDocument, fileName: string, languageId: string) {
```

Comment:
if not a quarkus project, we should bail here

Reply:
I think we should call `ProjectLabelInfo.getProjectLabelInfo` at the end of the method because it can takes some times.

Reply:
If preference is set to `ignore`, then no need to do the check, just bail.
Else, we need to perform the quarkus check *before* we ask the user. Else, if you open a spring-boot project, you'll be asked to switch the file's language, then check if it quarkus and do nothing, which make no sense

Reply:
> If preference is set to ignore, then no need to do the check, just bail.

fixed

> if you open a spring-boot project, you'll be asked to switch the file's language, then check if it's quarkus and do nothing, which makes no sense

Good catch, fixed